src/test/java/uk/gov/pay/api/it/validation/PaymentResourceMetadataValidationFailuresITest.java

```diff
@@ -0,0 +1,163 @@
+package uk.gov.pay.api.it.validation;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.restassured.response.ValidatableResponse;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import junitparams.converters.Nullable;
+import org.apache.http.HttpStatus;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import uk.gov.pay.api.it.PaymentResourceITestBase;
+import uk.gov.pay.api.utils.JsonStringBuilder;
+import uk.gov.pay.api.utils.mocks.CreateChargeRequestParams;
+import uk.gov.pay.api.utils.mocks.CreateChargeRequestParams.CreateChargeRequestParamsBuilder;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static java.util.Arrays.asList;
+import static javax.ws.rs.core.HttpHeaders.AUTHORIZATION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.core.Is.is;
+import static uk.gov.pay.api.it.CreatePaymentITest.paymentPayload;
+import static uk.gov.pay.api.model.TokenPaymentType.CARD;
+import static uk.gov.pay.api.utils.mocks.CreateChargeRequestParams.CreateChargeRequestParamsBuilder.aCreateChargeRequestParams;
+
+@RunWith(JUnitParamsRunner.class)
+public class PaymentResourceMetadataValidationFailuresITest extends PaymentResourceITestBase {
+
+    private static CreateChargeRequestParamsBuilder createChargeRequestParamsBuilder = aCreateChargeRequestParams()
+                .withAmount(100)
+                .withDescription("DESCRIPTION")
+                .withReference("REFERENCE")
+                .withReturnUrl("https://somewhere.gov.uk/rainbow/1");
+    
+    @Before
+    public void before() {
+        publicAuthMock.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID, CARD);
+    }
+
+    @Test
+    public void valueIsNotAStringBooleanOrNumber() {
+
+        CreateChargeRequestParams createChargeRequestParams = createChargeRequestParamsBuilder
+                .withMetadata(Map.of("foo", List.of("cake", "chocolate"), "bar", Map.of("a", "b")))
+                .build();
+
+        assertMetadataValidationError(createChargeRequestParams, 
+                "Invalid attribute value: Field [metadata] values must be of type String, Boolean or Number");
+    }
+
+    @Test
+    @Parameters({"", " ", "keyMoreThanThirtyCharskeyMoreThanThirtyCharskeyMoreThanThirtyChars"})
```

Comment:
Maybe show `@Parameters` at a team tech meeting, looks useful but I don't see it being widely used.

Reply:
it's used in connector too!